### DISCUSSION TITLE
Condition and Arguments seem not to be same evaluated

### DISCUSSION THREAD

**Discussion by dev-ivan-markovic** [axuno/SmartFormat#188] [Q&A] [answered]
Hello everybody,

I have an issue with the following logic:
`		 var input = "{\"FirstName\": \"MyName\",\"LastName\": \"MyLastName\",\"OrderNumber\": \"Test214\"}";
         JObject parsed = JObject.Parse(input);
         var result = Smart.Format("TEST: {OrderNumber:True {OrderNumber} |False {OrderNumber}}", parsed);
		 Console.WriteLine(result);`
Condition is always false. Do you see something wrong with this approach?

Please check the example here:
https://dotnetfiddle.net/zJ6BfN

Thank you
Ivan

**Comment by axunonb** [comment, marked as answer]
Hi Ivan
From the format string it's hard to figure out, what you would like to achieve: What does `True` or `False` in terms of the `OrderNumber`mean? I guess you would like to use the `ChooseFormatter` or the `ConditionalFormatter`. Did you take a look at the samples in the [Wiki](https://github.com/axuno/SmartFormat/wiki)?
If you change your example to
```CSharp
var result = Smart.Format("TEST: {OrderNumber:choose(Test214):{}|other number}", parsed);
```
you will get "other number" in the ouput, if the `OrderNumber` is not "Test214".
Hope this helps to get started.

**Reply by dev-ivan-markovic** [reply]
Hi @axunonb,

Thank you for checking this out.

Actually I want simply: string.IsNullOrEmpty(OrderNumber)? "TrueString" : ""

This logic I am using in HTML template, where I want to printout label if I have OrderNumber value (not empty or null), if not, don't show label.

Example from my test:

```csharp
 
var jsonInput = "{\"FirstName\": \"MyName\",\"LastName\": \"MyLastName\",\"OrderNumber\": \"Test214\"}"; 
JObject parsedObject = JObject.Parse(input); 
var result = Smart.Format("{OrderNumber: 'TrueString' | '' }", parsedObject);
Console.WriteLine(result);

**Reply by dev-ivan-markovic** [reply]
Something like this would be ideal:

```csharp
var result = Smart.Format("{OrderNumber:choose(null):{}| sometext}", parsed);

**Reply by axunonb** [reply]
For this case, use the [ConditionalFormatter](https://github.com/axuno/SmartFormat/wiki/Conditional), with a modified example form the Wiki:
```CSharp
var input = new { OrderNumber = "Order123" };
var result = Smart.Format("{OrderNumber:cond:{}|No text to display}", input);
Console.WriteLine(result);
```
If `OrderNumber` is null or empty, the text literal will be rendered.

**Reply by dev-ivan-markovic** [reply]
This works like a charm with anonymous objects. I have a problem with my json logic.
```csharp
var inputFromJson = JObject.Parse("{OrderNumber:null}");
var result = Smart.Format("{OrderNumber:cond:Exists|EmptyText}", inputFromJson);
```
The result is "Exists" :(
If you have idea please, if not I will search a way to make anonymous from JSON without knowing a definition of anonymous. 

Thank you!

**Reply by axunonb** [reply]
`var result = Smart.Format("{OrderNumber.ToString:cond:Exists|EmptyText}", inputFromJson);`

**Reply by dev-ivan-markovic** [reply]
That's it! Beautiful.

Thank you very much!
